File: Components/WinForms/VB/SchedulerMainDemo/Modules/Reporting/TimelineStyle/Report.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraScheduler.Reporting

Namespace DevExpress.XtraScheduler.Demos.Reporting.TimelineStyle
	''' <summary>
	''' Summary description for Report.
	''' </summary>
	Partial Public Class Report
		Inherits XtraSchedulerReport
		Private firstTimeScale As String ="Year"
		Private secondTimeScale As String = "Month"

		Public Sub New()

			InitializeComponent()
		End Sub
		Public Property VisibleIntervalCount() As Integer
			Get
				Return reportTimelineView1.VisibleIntervalCount
			End Get
			Set(ByVal value As Integer)
				reportTimelineView1.VisibleIntervalCount = value
			End Set
		End Property
		Public Property VisibleResourceCount() As Integer
			Get
				Return reportTimelineView1.VisibleResourceCount
			End Get
			Set(ByVal value As Integer)
				reportTimelineView1.VisibleResourceCount = value
			End Set
		End Property
		Public ReadOnly Property Scales() As TimeScaleCollection
			Get
				Return reportTimelineView1.Scales
			End Get
		End Property
		Public Property FirstLevelTimeScale() As String
			Get
				Return firstTimeScale
			End Get
			Set(ByVal value As String)
				If firstTimeScale = value Then
					Return
				End If
				firstTimeScale = value
				UpdateTimeScales()
			End Set
		End Property
		Public Property SecondLevelTimeScale() As String
			Get
				Return secondTimeScale
			End Get
			Set(ByVal value As String)
				If secondTimeScale = value Then
					Return
				End If
				secondTimeScale = value
				UpdateTimeScales()
			End Set
		End Property

		Private Sub UpdateTimeScales()
			Dim scales As TimeScaleCollection = reportTimelineView1.Scales
			scales.BeginUpdate()
			Try
			For Each item As TimeScale In scales
				item.Enabled = (item.DisplayName = firstTimeScale) OrElse (item.DisplayName = secondTimeScale)
			Next item
			Finally
			scales.EndUpdate()
			End Try
		End Sub
	End Class
End Namespace

File: Components/WinForms/CS/SchedulerMainDemo/Modules/Reporting/TimelineStyle/Report.cs
using System;
using DevExpress.XtraScheduler.Reporting;

namespace DevExpress.XtraScheduler.Demos.Reporting.TimelineStyle
{
	/// <summary>
	/// Summary description for Report.
	/// </summary>
	public partial class Report : XtraSchedulerReport {
        string firstTimeScale ="Year";
        string secondTimeScale = "Month";

		public Report() {

			InitializeComponent();
		}
		public int VisibleIntervalCount {
			get { return reportTimelineView1.VisibleIntervalCount; }
            set { reportTimelineView1.VisibleIntervalCount = value; }
		}
        public int VisibleResourceCount {
            get { return reportTimelineView1.VisibleResourceCount; }
            set { reportTimelineView1.VisibleResourceCount = value; }
        }
        public TimeScaleCollection Scales {
            get { return reportTimelineView1.Scales; } 
        }
        public string FirstLevelTimeScale {
            get {
                return firstTimeScale; 
            }
            set {
                if (firstTimeScale == value)
                    return;
                firstTimeScale = value;
                UpdateTimeScales();
            } 
        }
        public string SecondLevelTimeScale {
            get {
                return secondTimeScale; 
            }
            set {
                if (secondTimeScale == value)
                    return;
                secondTimeScale = value;
                UpdateTimeScales();
            } 
        }

        private void UpdateTimeScales() {
            TimeScaleCollection scales = reportTimelineView1.Scales;
            scales.BeginUpdate();
            try {
            foreach (TimeScale item in scales)
                item.Enabled = (item.DisplayName == firstTimeScale) || (item.DisplayName == secondTimeScale);
            } finally {
            scales.EndUpdate();
            }
        }
	}
}
